platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformSchedulers.java

```diff
@@ -50,7 +51,9 @@ public record PlatformSchedulers(
         @NonNull TaskScheduler<List<ConsensusRound>> linkedEventIntakeScheduler,
         @NonNull TaskScheduler<GossipEvent> eventCreationManagerScheduler,
         @NonNull TaskScheduler<StateSavingResult> signedStateFileManagerScheduler,
-        @NonNull TaskScheduler<StateSignatureTransaction> stateSignerScheduler) {
+        @NonNull TaskScheduler<StateSignatureTransaction> stateSignerScheduler,
+        @NonNull TaskScheduler<Void> applicationTransactionPrehandlerScheduler,
+        @NonNull TaskScheduler<Void> systemTransactionPrehandlerScheduler) {
```

Comment:
why have a separate handler for system transactions?

Reply:
I acknowledge this is ugly, but currently our system transaction handling is not thread safe. Current system transactions are performed on a `QueueThread`.

This is very much a short term solution that I expect to see improved in the near future. Once we get system transaction handling onto the wiring framework, we can do this a lot more elegantly. Instead of having a component that handles all system transactions, we can just ensure that the transactions end up on the proper wire.

Reply:
as discussed, this second scheduler will become the signature collector